### PR TITLE
Add alternatives

### PR DESCRIPTION
**Pull request by jbarnette**
This tidies up the "Alternatives" section a bit and adds the projects mentioned in @daylerees' https://github.com/github/scientist/pull/35.

@jesseplusplus @rick @zerowidth I haven't audited these for licenses or completeness at all, any concern? I'll take a quick look before this ships.

/cc @Haacked @joealcorn @MadcapJake @rawls238 @tomiaijo @tcrayford @ziyasal @lancew

### CODE REVIEW

**Comment by natesholland** [comment]
I checked the licenses, they are all MIT except [MadcapJake/Test-Lab](https://github.com/MadcapJake/Test-Lab/blob/master/LICENSE) which is under the artistic license and [yeller/laboratory](https://github.com/yeller/laboratory/blob/master/LICENSE) which is under the Eclipse Public License. [lancew/Scientist](https://github.com/lancew/Scientist) has MIT in the readme but no specific LICENSE file. I don't know if that matters from a legal perspective.

**Comment by rawls238** [comment]
:+1:

File: README.md [hunk 1]
```diff
@@ -435,15 +435,23 @@ end
...
+- [haacked/scientist.net](https://github.com/haacked/scientist.net) (.NET)
+- [joealcorn/laboratory](https://github.com/joealcorn/laboratory) (Python)
+- [MadcapJake/Test-Lab](https://github.com/MadcapJake/Test-Lab) (Perl 6)
+- [rawls238/Scientist4J](https://github.com/rawls238/Scientist4J) (Java)
+- [tomiaijo/scientist](https://github.com/tomiaijo/scientist) (C++)
+- [trello/scientist](https://github.com/trello/scientist) (node.js)
+- [yeller/laboratory](https://github.com/yeller/laboratory) (Clojure)
+- [lancew/Scientist](https://github.com/lancew/Scientist) (Perl 5)
+- [lancew/ScientistP6](https://github.com/lancew/ScientistP6) (Perl 6)
+- [ziyasal/scientist.js](https://github.com/ziyasal/scientist.js) (node.js, ES6)
```
Comment (rawls238) [inline README.md]: maybe group them by language (i.e. all the javascript ones next to each other, all the perl ones next to each other, etc.)?

**Comment by jesseplusplus** [comment]
> @jesseplusplus @rick @zerowidth I haven't audited these for licenses or completeness at all, any concern? I'll take a quick look before this ships.

I'm on the fence about whether any version that hasn't really been completed or tested in the wild should be listed here, but I realize that's a bit more work to do that sort of audit.

**Comment by jbarnette** [comment]
> I'm on the fence

@jesseplusplus Same.

**Comment by natesholland** [comment]
Not a maintainer, but I agree with @jesseplusplus

**Comment by tcrayford** [comment]
ftr the clojure library (yeller/laboratory) has been used for a few hundred experiments in production, I'd say it's completed/ready.

**Comment by rick** [comment]
Agree w/ @jesseplusplus.

There are generally a few categories:

 1. This is clearly production-ready, supported, and in wide use
 2. This is pretty clearly not
 3. This might be, it's hard to tell

For the 3's, it seems like we're kind of looking for assurances that a thing is in use and supported. One approach is to only list the 1's. Another would be to get someone to vouch / act as contact for the 3's. A third approach would be to list the 3's in a separate section.

**Comment by Haacked** [comment]
> I'm on the fence about whether any version that hasn't really been completed or tested in the wild should be listed here, but I realize that's a bit more work to do that sort of audit.

Also, there's probably two audiences we're talking about here. One is looking to use a battle-tested Scientist for their platform. The other is looking to contribute to Scientist for their platform. I think we have an opportunity to help both out.

So maybe we could have two sections: "Battle Tested" (or better term) and "In Development". Probably doesn't make sense for us (GitHub) to verify, but we could trust members of the community to vouch for implementations until proven wrong.

**Comment by rawls238** [comment]
fwiw I kind of like how Facebook's PlanOut does this: http://facebook.github.io/planout/docs/ports-overview.html

**Comment by daylerees** [comment]
Completely understand your caution. My variation has the download count listed in the readme, a full test suite (100% coverage, although we know that's not as fancy as it sounds!), and I've used it in production on a couple of projects. I hope that helps!

**Comment by lancew** [comment]
Perhaps add a YMMV caveat to the list about maturity of the alternatives?

**Comment by lancew** [comment]
@natesholland lancew/scientist has the MIT in the module itself Scientist.pm and the POD documentation file.

**Comment by joealcorn** [comment]
The [python version](joealcorn/laboratory) is MIT licensed and I've been using it in production since day 1, I consider it prod-ready

**Comment by rawls238** [comment]
The Java one is production-ready though I agree that there should be a YMMV disclaimer on the alternatives

**Comment by rawls238** [comment]
Bump @jbarnette any updates on what you guys want to do with this?

**Comment by jbarnette** [comment]
@rawls238 Thanks for bumping this, I needed it. Top of my guilt list now, I'll get this spruced up and shipped today or tomorrow.

**Comment by rawls238** [comment]
Another bump...

**Comment by Haroenv** [comment]
Any reason why this hasn't been merged after all?

**Comment by jesseplusplus** [comment]
@jbarnette I'm gonna merge this. I tidied up the ordering, but didn't bother with any grouping. Deciding what should be considered a prod-ready can be left as an exercise to the user 😄

**Pull request merged**